docs/wip-plugins.md

```diff
@@ -149,41 +156,175 @@ c2.register_callback(
+The values behind the names may change, do not count on them. It has the
+following keys:
+
+- `Twitch`
+
+#### `ChannelType` enum
+
+This table describes channel types Chatterino supports. The values behind the
+names may change, do not count on them. It has the following keys:
+
+- `None`
+- `Direct`
+- `Twitch`
+- `TwitchWhispers`
+- `TwitchWatching`
+- `TwitchMentions`
+- `TwitchLive`
+- `TwitchAutomod`
+- `TwitchEnd`
+- `Irc`
+- `Misc`
+
+#### `Channel`
+
+This is a type that represents a channel. Existence of this object doesn't
+force Chatterino to hold the channel open. Should the user close the last split
+holding this channel open, your Channel object will expire. You can check for
+this using the `Channel:is_valid()` function. Using any other function on an
+expired Channel yields an error. Using any `Channel` member function on a
+non-`Channel` table also yields an error.
+
+Some functions make sense only for Twitch channel, these yield an error when
+used on non-Twitch channels. Special channels while marked as
+`is_twitch_channel() = true` do not have these functions. To check if a channel
+is an actual Twitch chatroom use `Channel:get_type()` instead of
+`Channel:is_twitch_channel()`.
+
+##### `Channel:by_name(name, platform)`
+
+Finds a channel given by `name` on `platform` (see [`Platform` enum](#Platform-enum)). Returns the channel or `nil` if not open.
+
+Some miscellaneous channels are marked as if they are specifically Twitch channels:
+
+- `/whispers`
+- `/mentions`
+- `/watching`
+- `/live`
+- `/automod`
+
+Example:
+
+```lua
+local pajladas = c2.Channel.by_name("pajlada", c2.Platform.Twitch)
+```
+
+##### `Channel:by_twitch_id(id)`
+
+Finds a channel given by the string representation of the owner's Twitch user ID. Returns the channel or `nil` if not open.
+
+Example:
+
+```lua
+local pajladas = c2.Channel.by_twitch_id("11148817")
+```
+
+##### `Channel:get_name()`
+
+On Twitch returns the lowercase login name of the channel owner. On IRC returns the normalized channel name.
+
+Example:
+
+```lua
+-- Note: if the channel is not open this errors
+pajladas:get_name()  -- "pajlada"
+```
+
+##### `Channel:get_type()`
+
+Returns the channel's type. See [`ChannelType` enum](#ChannelType-enum).
 
-Sends a message to `channel` with the specified text. Also executes commands.
+##### `Channel:get_display_name()`
+
+Returns the channel owner's display name. This can contain characters that are not lowercase and even non-ASCII.
 
 Example:
 
 ```lua
-function cmdShout(ctx)
+local saddummys = c2.Channel.by_name("saddummy")
+saddummys:get_display_name() -- "서새봄냥"
+```
+
+<!-- F Korean Twitch, apparently you were not profitable enough -->
+
+##### `Channel:send_message(message[, execute_commands])`
+
+Sends a message to the channel with the given text. If `execute_commands` is
+not present or `false` commands will not be executed client-side, this affects
+all user commands and all Twitch commands except `/me`.
```

Comment:
I think the examples below are inconsistent with this description. "If `execute_commands` is not present [...] commands will not be executed client-side [...]" contradicts that for example
```lua
pajladas:send_message("hello")
```
will execute the command.